docs/rules/no-implicit-injections.md

```diff
@@ -0,0 +1,112 @@
+# ember/no-implicit-injections
+
+🔧 This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).
+
+<!-- end auto-generated rule header -->
+
+Ember 3.26 introduced a deprecation for relying on implicit service injections or allowing addons to implicitly inject services into all classes of certain types. Support for this is dropped in Ember 4.0.
+
+In many applications, `this.store` from Ember Data is often used without injecting the `store` service in Controllers or Routes. Other addons may also have included implicit service injections via initializers and the `application.inject` API.
+
+To resolve this deprecation, a service should be explicitly declared and injected using the [service injection decorator](https://api.emberjs.com/ember/3.28/functions/@ember%2Fservice/inject).
+
+## Rule Details
+
+This rule checks for a configured list of previously auto injected services and warns if they are used in classes without explicit injected service properties.
+
+## Examples
+
+Examples of **incorrect** code for this rule:
+
+```js
+// routes/index.js
+import Route from '@ember/routing/route';
+
+export default class IndexRoute extends Route {
+  model() {
+    return this.store.findAll('rental');
+  }
+}
+
+```
+
+```js
+// controllers/index.js
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+
+export default class IndexController extends Controller {
+  @action
+  loadUsers() {
+    return this.store.findAll('user');
+  }
+}
+```
+
+Examples of **correct** code for this rule:
+
+```js
+// routes/index.js
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class IndexRoute extends Route {
+  @service('store') store;
+
+  model() {
+    return this.store.findAll('rental');
+  }
+}
+```
+
+```js
+// controller/index.js
+import Route from '@ember/routing/route';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default class IndexController extends Controller {
+  @service('store') store;
+
+  @action
+  loadUsers() {
+    return this.store.findAll('user');
+  }
+}
+```
+
+## Migration
+
+The autofixer for this rule will update classes and add injections for the configured services.
+
+## Configuration
+
+This lint rule will search for instances of `store` used in routes or controllers by default. If you have other services that you would like to check for uses of, the configuration can be overridden.
+
+- object -- containing the following properties:
+  - array -- `services` -- Array of configuration objects configuring the lint rule to check for use of implicit injected services
```

Comment:
One observation is that [ember/no-restricted-service-injections](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-restricted-service-injections.md#configuration) accepts an unnamed, top-level array of objects for its schema, each with a `services` property.

I actually think what you have with a named, top-level array property is probably better practice since it allows us to add other top-level, named properties as needed.

But one way we could possibly improve yours is to name the top-level property `denylist` instead of `services` to avoid confusion between `services` and `serviceNames`. We could also optionally name the nested property `services` instead of `serviceNames` for consistency with the other rule, but I don't have a strong preference between these names.

Reply:
Updated this to `denyList` and `service`